source/mesh_deformation/external_tool_interface.cc

```diff
@@ -0,0 +1,518 @@
+         * describe the properties of each datum.
+         * @param [in] data_component_interpretations An optional vector that
+         * controls if the properties are interpreted as scalars, vectors,
+         * or tensors. Has to be of the same length as @p data_component_names.
+         */
+        void
+        build_patches(const std::vector<Point<spacedim>> &locations,
+                      const std::vector<std::vector<double>> &data = {},
+                      const std::vector<std::string> &data_component_names = {},
+                      const std::vector<
+                      DataComponentInterpretation::DataComponentInterpretation>
+                      &data_component_interpretations_ = {})
+        {
+          Assert(
+            data_component_names.size() == data_component_interpretations_.size(),
+            ExcMessage(
+              "When calling Particles::DataOut::build_patches with data component "
+              "names and interpretations you need to provide as many data component "
+              "names as interpretations. Provide the same name for components that "
+              "belong to a single vector or tensor."));
+
+          dataset_names.clear();
+          dataset_names.emplace_back("id");
+          dataset_names.insert(dataset_names.end(),
+                               data_component_names.begin(),
+                               data_component_names.end());
+
+          data_component_interpretations.clear();
+          data_component_interpretations.emplace_back(
+            DataComponentInterpretation::component_is_scalar);
+          data_component_interpretations.insert(
+            data_component_interpretations.end(),
+            data_component_interpretations_.begin(),
+            data_component_interpretations_.end());
+
+          const unsigned int n_property_components = data_component_names.size();
+          const unsigned int n_data_components     = dataset_names.size();
+
+          patches.resize(locations.size());
+
+          for (unsigned int i = 0; i < locations.size(); ++i)
+            {
+              patches[i].vertices[0] = locations[i];
+              patches[i].patch_index = i;
+
+              // We have one more data components than dataset_names (the particle id)
+              patches[i].data.reinit(n_data_components, 1);
+
+              patches[i].data(0, 0) = i; // id
+
+              if (n_data_components > 1)
+                {
+                  for (unsigned int property_index = 0;
+                       property_index < n_property_components;
+                       ++property_index)
+                    patches[i].data(property_index + 1, 0) =
+                      data[i][property_index];
+                }
+            }
+        }
+
+      protected:
+        /**
+         * Returns the patches built by the data_out class which was previously
+         * built using a particle handler
```

Comment:
Not your fault, but let's get it right grammatically (see also https://github.com/dealii/dealii/pull/18867)
```suggestion
       * Return the patches previously built by the build_patches() function.
```